### ISSUE TITLE
Values of constans wrapped in double quotes breaks JSHint

### ISSUE THREAD

**Issue by jayrmotta** [ajwhite/gulp-ng-config#12]
A part of the strings on this file are written in single quotes and the other part is written in double quotes and it breaks JSHint validation.

Example:
```
angular.module('foo.config', [])
.constant('foo', "bar");
```

**Comment by ajwhite** [comment]
Whoops! thanks

**Comment by ajwhite** [comment]
For now, may be good to ignore the generated file in your `jshint` task.

**Comment by jayrmotta** [comment]
@ajwhite that's fine!

Here is my workaround until we fix it:

```javascript
gulp.task('config', function() {
  var env = argv.env ? argv.env : 'development';

  gulp.src('conf/' + env + '.json')
  .pipe(gulpNgConfig('cockpitUi.config', {
    wrap: '(function () {\n\'use strict\';\n/*jshint ignore:start*/\n return <%= module %> /*jshint ignore:end*/\n})();'
  }))
  .pipe(rename('config.js'))
  .pipe(gulp.dest('src/app/'));
});
```

Just ignore the rename and env hack, I did that so I could change among many configs easily.

**Comment by jasonswett** [comment]
+1 for defaulting to single quotes. @jayrmotta's wrap solution worked for me.

**Comment by ajwhite** [comment]
Placing on the roadmap

**Comment by ajwhite** [comment]
Fixed in #29 by @psalaets :+1:

**Comment by ajwhite** [comment]
Actually, just realized the PR was for double quotes, which broke your tests.

Going to reopen for further consideration. A possibility would be to configure `singleQuotes: Boolean`

**Comment by johannesjo** [comment]
I would be OK with either one. It doesn't really matter. Just would be nice to get rid of the warning. singleQuotes: Boolean has the advantage that it would work not only with jshint but also with jscs.

**Comment by wembernard** [comment]
Would it be that hard to transform "strings" to 'strings' during the JSON to JS process?

**Comment by ajwhite** [comment]
@wembernard yep, we'll be going with the `singleQuotes` mode. It should not be that hard, just some escaping required. Feel free to open a PR

**Comment by wembernard** [comment]
@ajwhite I forked your project and managed to handle this by changing

    _.each(jsonObj, function (value, key) {
      constants.push({
        name: key,
        value: JSON.stringify(value, null, spaces)
      });
    });

to

    _.each(jsonObj, function (value, key) {
      var processedValue;

      if (typeof value === 'boolean' || typeof value === 'number') {
        processedValue = value;
      } else {
        processedValue = JSON.stringify(value, null, spaces).replace(/"/g, '\'');
      }

      constants.push({
        name: key,
        value: processedValue
      });
    });

However, I can't find a way to pass your ```gulp test```.

**Comment by ajwhite** [comment]
The outputs have changed, so all the tests will fail. This should only happen if single quote property is provided

**Comment by elporfirio** [comment]
exist a method or function to change a singlequote after gulp-ng-config?

**Comment by markmssd** [comment]
Sounds pretty good to me! GJ!

**Comment by Torone** [comment]
Hi, any news about this? I don't think an option for `singleQuote` has been added and my generated file has double quote yet. Thanks

**Comment by franbueno** [comment]
Any news? Thanks! :)

**Comment by Torone** [comment]
Well, I just created a plugin that do the dirty job. You can find it here: [https://www.npmjs.com/package/gulp-replace-quotes](https://www.npmjs.com/package/gulp-replace-quotes). It replace single quotes to double quotes or vice versa...
